.pre-commit-config.yaml

```diff
@@ -0,0 +1,44 @@
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.5.0
+    hooks:
+      - id: trailing-whitespace
+        files: '\.py$'
+      - id: end-of-file-fixer
+        files: '\.py$'
+      - id: check-yaml
+      - id: check-added-large-files
+        files: '\.py$'
+
+  - repo: https://github.com/psf/black
+    rev: 24.3.0
+    hooks:
```

Comment:
Black should fix trailing whitespace and I think there is an end-of-file setting... or maybe that's VSCode I'm thinking of.